libkbfs/delayed_cancellation.go

```diff
@@ -192,17 +186,18 @@ func NewContextWithCancellationDelayer(
 // delayed cancellation for ctx. This is useful to indicate that the
 // operation(s) associated with the context has entered a critical state, and
 // it should not be canceled until after timeout or CleanupCancellationDelayer
-// is called.
+// is called. Note that if EnableDelayedCancellationWithGracePeriod is called
+// for the second time, and the grace period has started due to a cancellation,
+// the grace period would not be extended (i.e. timeout has no effect in this
+// case).
```

Comment:
Maybe add something like ", although no error will be returned" at the end of the sentence?  It's a little surprising that the new timeout will be ignored without an error being thrown.